pkgs/applications/editors/emacs-modes/melpa-packages.nix

```diff
@@ -207,6 +207,17 @@ self:
       # upstream issue: missing file header
       qiita = markBroken super.qiita;
 
+      racer =
+        (super.racer.overrideAttrs (attrs: {
+          patches =
+            (attrs.patches or []) ++ [
+              (external.substituteAll {
+                src = ./racer-mode/rust-src.patch;
+                inherit (external.rustPlatform) rustcSrc;
+              })
+            ];
+        }));
```

Comment:
I suggest this instead (untested, but should work):

```nix
{
  racer = super.racer.overrideAttrs (attrs: {
    postPatch = attrs.postPatch or "" + ''
      substituteInPlace racer.el \
        --replace /usr/local/src/rust/src ${external.rustPlatform.rustcSrc}
    '';
  });
}
```

This is simpler and more resiliant to changes upstream.

Reply:
Done